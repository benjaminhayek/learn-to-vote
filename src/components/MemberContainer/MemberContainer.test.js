import React from 'react';
import { MemberContainer } from './index';
import { shallow } from 'enzyme';

describe('MemberContainer', () => {
  let wrapper;
  let congressmen;
  let bills;
  
  beforeEach(() => {
    bills = [{title: 'Bill1', committee: 'House', url:'www.gov.com'}];
    congressmen = [{name: 'Ted', id: 1, party: 'D', title: 'Representative', nextElection: '2018', selected: false}, {name: 'Jan', id: 2, party: 'R', title: 'Representative', nextElection: '2018', selected: false}];    
    wrapper = shallow(<MemberContainer congressmen={congressmen}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have an initial state of bills set to an empty array', () => {
    expect(wrapper.state().bills).toEqual([]);
  });

  it('should set the state of bills when handleSubmit is envoked', () => {
    const expected = []
    wrapper.instance().handleSubmit();
    expect(wrapper.state().bills).toEqual(expected);
  });

  it('should reset the state on click', () => {
    const expected = []
    wrapper.instance().resetState()
    expect(wrapper.state().bills).toEqual(expected)
  })
})