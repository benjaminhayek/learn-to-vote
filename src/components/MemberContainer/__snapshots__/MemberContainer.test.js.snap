// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberContainer should match the snapshot 1`] = `
<div>
  <div>
    <h1
      className="container-title"
    >
      Your Congressmen
    </h1>
    <h1
      className="card-container"
    >
      <Connect(MemberCard)
        congressmen={
          Object {
            "id": 1,
            "name": "Ted",
            "nextElection": "2018",
            "party": "D",
            "selected": false,
            "title": "Representative",
          }
        }
        key="f18dd2c7-0249-46ec-818d-ee98090fc8bc"
      />
      <Connect(MemberCard)
        congressmen={
          Object {
            "id": 2,
            "name": "Jan",
            "nextElection": "2018",
            "party": "R",
            "selected": false,
            "title": "Representative",
          }
        }
        key="53a64be8-42a6-4412-b337-80d8d9535bc6"
      />
    </h1>
  </div>
</div>
`;
